packages/expo-cli/src/commands/generate-module/TemplateFetcher.js

```diff
@@ -0,0 +1,66 @@
+import path from 'path';
+import fse from 'fs-extra';
+import child_process from 'child_process';
+
+const NPM_TEMPLATE_VERSION = '2.0.0';
+const NPM_TEMPLATE = 'expo-module-template';
+
+/**
+ * Decompresses tarball archive obtained from npm
+ * @param {path} archive - path to actual archive
+ * @param {path} destinationDir - destination for uncompressed files
+ */
+const decompress = async (archive, destinationDir) => {
+  return new Promise((resolve, reject) => {
+    targz.decompress(
```

Comment:
See also [`pacote.extract`](https://github.com/zkat/pacote/blob/latest/README.md#extract), which can fetch the package from npm and extract it, all at once, without having to spawn npm. `pacote` is used internally by the npm CLI and we also use it for `expo init` templates.

Since pacote supports all kinds of package specifiers supported by `npm install`, you can also eliminate the `--template-directory` option and only allow `--template` (`expo-module-template` by default) so the user can pass `--template <local-path-here)` to use a local template or `--template <npm-package>` to use a package from npm.